src/kepler_model/estimate/model_server_connector.py

```diff
@@ -8,22 +8,31 @@
 from kepler_model.util.config import is_model_server_enabled, get_model_server_req_endpoint, get_model_server_list_endpoint, download_path
 from kepler_model.util.loader import get_download_output_path
 from kepler_model.util.train_types import ModelOutputType
+from kepler_model.train.profiler.node_type_index import discover_spec_values
 
+machine_spec_mount_path = "/etc/kepler/models/machine_spec"
+machine_id = os.getenv('MACHINE_ID', None)
 
-# discover_spec: determine node spec in json format (refer to NodeTypeSpec)
-def discover_spec():
-    import psutil
+# get_spec_values: determine node spec in json format (refer to NodeTypeSpec)
+def get_spec_values(machine_id : str|None):
+    if machine_id is not None:
+        spec_file = os.path.join(machine_spec_mount_path, machine_id)
+        try:
+            with open(spec_file) as f:
+                res = json.load(f)
+            return res
+        except:
+            pass
+    return discover_spec_values()
 
-    # TODO: reuse node_type_index/generate_spec with loosen selection
-    cores = psutil.cpu_count(logical=True)
-    spec = {"cores": cores}
-    return spec
-
-
-node_spec = discover_spec()
+node_spec = None
 
 
 def make_model_request(power_request):
+    global node_spec
+    if node_spec is None:
+         node_spec = get_spec_values(machine_id)
+         print(f"Node spec: {node_spec}")
```

Comment:
lets make use of `logging` to log or `click.echo` to print to console. This allows us (and `ruff`) to weed out debug messages from intentional console output.